src/client/datascience/jupyter/kernels/kernelSelector.ts

```diff
@@ -535,6 +535,10 @@ export class KernelSelector implements IKernelSelectionUsage {
 
         // First use our kernel finder to locate a kernelspec on disk
         const kernelSpec = await this.kernelFinder.findKernelSpec(resource, notebookMetadata, cancelToken);
+        traceInfoIf(
+            !!process.env.VSC_JUPYTER_FORCE_LOGGING,
+            `Kernel spec found ${JSON.stringify(kernelSpec)}, metadata ${JSON.stringify(notebookMetadata || '')}`
```

Comment:
Useful on CI.
Found that C# kernels do not work when no jupyter and tests were failing, this helped.
Leaving for now, as its useful (for CI).